deps/rabbit/src/rabbit_stream_queue.erl

```diff
@@ -1111,12 +1112,12 @@ binary_to_msg(#resource{kind = queue,
     %% If exchange or routing_keys annotation isn't present the data most likely came
     %% from the rabbitmq-stream plugin so we'll choose defaults that simulate use
     %% of the direct exchange.
-    Mc = case mc:get_annotation(exchange, Mc0) of
-             undefined -> mc:set_annotation(exchange, <<>>, Mc0);
+    Mc = case mc:get_annotation(?EXCHANGE, Mc0) of
```

Comment:
Given we have now exchange and routing key apis in `mc` could we do the defaulting in there instead so that it always returns a result ifyou call, e.g. `mc:exchange(Msg)`?